addons/pkg-npm/jaxrs/src/main/java/org/commonjava/indy/pkg/npm/jaxrs/NPMContentAccessHandler.java

```diff
@@ -0,0 +1,466 @@
+                                                  Response response,
+                                                  final Consumer<Response.ResponseBuilder> builderModifier )
+    {
+        if ( transfer == null )
+        {
+            return;
+        }
+
+        OutputStream out1 = null;
+        OutputStream out2 = null;
+
+        Transfer target1;
+        Transfer target2;
+        String content1 = "";
+        String content2 = "";
+
+        ConcreteResource resource = transfer.getResource();
+        try
+        {
+            ObjectMapper mapper = new ObjectMapper();
+            JsonNode root = mapper.readTree( transfer.openInputStream( true ) );
+
+            String path1 = null;
+            String path2 = null;
+            JsonNode vnode = root.path( "versions" );
+            JsonNode anode = root.path( "_attachments" );
+            JsonNode idnode = root.path( "_id" );
+
+            if ( vnode.fields().hasNext() )
+            {
+                String version = vnode.fields().next().getKey();
+                if ( version == null )
+                {
+                    return;
+                }
+                path1 = Paths.get( idnode.asText(), version ).toString();
+                content1 = vnode.findValue( version ).toString();
+            }
+
+            if ( anode.fields().hasNext() )
+            {
+                String tarball = anode.fields().next().getKey();
+                if ( tarball == null )
+                {
+                    return;
+                }
+                path2 = Paths.get( idnode.asText(), "-", tarball ).toString();
+                content2 = anode.findPath( "data" ).asText();
+            }
+
+            if ( path1 == null || path2 == null )
+            {
+                return;
+            }
+            target1 = transfers.getCacheReference( new ConcreteResource( resource.getLocation(), path1 ) );
+            logger.info( "STORE {}", target1.getResource() );
+
+            target2 = transfers.getCacheReference( new ConcreteResource( resource.getLocation(), path2 ) );
+            logger.info( "STORE {}", target2.getResource() );
+        }
+        catch ( final IOException e )
+        {
+            logger.error( String.format( "[NPM] Json node parse failed for resource: %s. Reason: %s", resource,
+                                         e.getMessage() ), e );
+            response = formatResponse( e, builderModifier );
+
+            return;
+        }
+        try
+        {
+            if ( target1 == null || target2 == null )
+            {
+                return;
+            }
+            out1 = target1.openOutputStream( TransferOperation.UPLOAD, true, eventMetadata );
+            out1.write( content1.getBytes() );
+
+            out2 = target2.openOutputStream( TransferOperation.UPLOAD, true, eventMetadata );
+            out2.write( Base64.decodeBase64( content2 ) );
+
+            setGeneratedTransfers( transfer, target1, target2 );
```

Comment:
Let's avoid to use so many xxx1 and xxx2 var names in a method which contains so many codes, especially when these vars will be used many times in this method.